src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,63 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
 public class SalaryInfo {
+    private static final int dateIndex = 0;
+    private static final int nameIndex = 1;
+    private static final int hoursWorkedIndex = 2;
+    private static final int hourlyRateIndex = 3;
+    private static final int expectedParts = 4;
+
+    private int calculateTotalIncome(String[] data, String name, Date fromDate, Date toDate,
+                                     SimpleDateFormat dateFormat) throws ParseException {
+        int totalIncome = 0;
+        for (String entry : data) {
+            String[] parts = entry.split(" ");
+            if (isValidEntry(parts, name, fromDate, toDate, dateFormat)) {
+                int hoursWorked = Integer.parseInt(parts[hoursWorkedIndex]);
+                int hourlyRate = Integer.parseInt(parts[hourlyRateIndex]);
+                totalIncome += hoursWorked * hourlyRate;
+            }
+        }
+        return totalIncome;
+    }
+
+    private boolean isValidEntry(String[] parts, String name, Date fromDate,
+                                 Date toDate, SimpleDateFormat dateFormat) throws ParseException {
+        if (parts.length == expectedParts) {
+            Date entryDate = dateFormat.parse(parts[dateIndex]);
+            return entryDate.compareTo(fromDate) >= 0 && entryDate.compareTo(toDate)
+                    <= 0 && parts[nameIndex].equals(name);
+        }
+        return false;
+    }
+
+    private String reportGenerator(String dateFrom, String dateTo) {
+        return "Report for period " + dateFrom + " - " + dateTo;
+    }
+
+    private String salaryLineGenerator(String name, int totalIncome) {
+        return name + " - " + totalIncome;
+    }
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder result = new StringBuilder();
+        result.append(reportGenerator(dateFrom, dateTo));
+        SimpleDateFormat dateFormat = new SimpleDateFormat("dd.MM.yyyy");
+        try {
+            Date fromDate = dateFormat.parse(dateFrom);
+            Date toDate = dateFormat.parse(dateTo);
```

Comment:
make sure to read this before submitting your task: [checklist](https://github.com/mate-academy/jv-salary-info/blob/master/checklist.md)